pyomo/contrib/solver/tests/solvers/test_highs.py

```diff
@@ -0,0 +1,158 @@
+        # Tests issue https://github.com/Pyomo/pyomo/issues/3127
+
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var(domain=pyo.Binary)
+        m.y = pyo.Var(domain=pyo.Binary)
+        m.fx = pyo.Var(domain=pyo.NonNegativeReals)
+        m.fy = pyo.Var(domain=pyo.NonNegativeReals)
+        m.c1 = pyo.Constraint(expr=m.fx <= m.x)
+        m.c2 = pyo.Constraint(expr=m.fy <= m.y)
+        m.c3 = pyo.Constraint(expr=m.x + m.y <= 1)
+
+        m.obj = pyo.Objective(expr=m.fx * 0.5 + m.fy * 0.4, sense=pyo.maximize)
+
+        opt = Highs()
+
+        # solution 1 has m.x == 1 and m.y == 0
+        r = opt.solve(m)
+        self.assertAlmostEqual(m.fx.value, 1, places=5)
+        self.assertAlmostEqual(m.fy.value, 0, places=5)
+        self.assertAlmostEqual(r.objective_bound, 0.5, places=5)
+
+        # solution 2 has m.x == 0 and m.y == 1
+        m.y.fix(1)
+        r = opt.solve(m)
+        self.assertAlmostEqual(m.fx.value, 0, places=5)
+        self.assertAlmostEqual(m.fy.value, 1, places=5)
+        self.assertAlmostEqual(r.objective_bound, 0.4, places=5)
+
+        # solution 3 should be equal solution 1
+        m.y.unfix()
+        m.x.fix(1)
+        r = opt.solve(m)
+        self.assertAlmostEqual(m.fx.value, 1, places=5)
+        self.assertAlmostEqual(m.fy.value, 0, places=5)
+        self.assertAlmostEqual(r.objective_bound, 0.5, places=5)
+
+    def test_qp1(self):
+        # test issue #3381
+        m = pyo.ConcreteModel()
+
+        m.x1 = pyo.Var(name='x1', domain=pyo.Reals)
+        m.x2 = pyo.Var(name='x2', domain=pyo.Reals)
+
+        # Quadratic Objective function
+        m.obj = pyo.Objective(expr=m.x1 * m.x1 + m.x2 * m.x2, sense=pyo.minimize)
+
+        m.con1 = pyo.Constraint(expr=m.x1 >= 1)
+        m.con2 = pyo.Constraint(expr=m.x2 >= 1)
+
+        results = opt.solve(m)
+        self.assertAlmostEqual(m.x1.value, 1, places=5)
+        self.assertAlmostEqual(m.x2.value, 1, places=5)
+        self.assertEqual(results.objective_bound, 2)
+
+    def test_qp2(self):
+        # test issue #3381
+        m = pyo.ConcreteModel()
+
+        m.x1 = pyo.Var(name='x1', domain=pyo.Reals)
+        m.x2 = pyo.Var(name='x2', domain=pyo.Reals)
+
+        m.p = pyo.Param(initialize=1, mutable=True)
+
+        m.obj = pyo.Objective(
+            expr=m.p * m.x1 * m.x1 + m.x2 * m.x2 - m.x1 * m.x2, sense=pyo.minimize
+        )
+
+        m.con1 = pyo.Constraint(expr=m.x1 >= 1)
+        m.con2 = pyo.Constraint(expr=m.x2 >= 1)
+
+        opt.set_instance(m)
+        results = opt.solve(m)
+        self.assertAlmostEqual(m.x1.value, 1, places=5)
+        self.assertAlmostEqual(m.x2.value, 1, places=5)
+        self.assertEqual(results.objective_bound, 1)
+
+        m.p.value = 2.0
+        opt.update_parameters()
+        results = opt.solve(m)
+        self.assertAlmostEqual(m.x1.value, 1, places=5)
+        self.assertAlmostEqual(m.x2.value, 1, places=5)
+        self.assertEqual(results.objective_bound, 2)
```

Comment:
I think these tests should be moved to `test_solvers.py`. Let's add a `qp_solvers` group [here](https://github.com/Pyomo/pyomo/blob/56521be56dadcbcd2d45ab95b95d9a0e1953e2ec/pyomo/contrib/solver/tests/solvers/test_solvers.py#L54) and apply these tests to all of the solvers in that group. For now, the group should include `gurobi_persistent`, `ipopt`, and `highs`.

Reply:
Done